vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/SlideManager.java

```diff
@@ -46,4 +54,36 @@ public ISlide getSlide(String slideId) {
     public void updateSlide(Slide slide) {
         slideRepo.save((Slide) slide);
     }
+    
+	@Override
+	public void deleteSlideById(String slideId, String moduleId, String flag) throws SlideDoesNotExistException {
+		try {
+
+			if (flag.equals("0")) {
```

Comment:
why do you need this?

Reply:
I am using the flag, to handle the condition when slide has any sequences.
When flag is 0 - The slide has no sequences.
Else, - It has sequences, and it deletes the references from those sequences.

Reply:
why do you need that? the else block takes care of that no matter what.

Reply:
But, then 
When I am deleting a slide that has no sequence, for the below code

List<Sequence> sequences = sequenceRepo.findSequencesForModule(moduleId);

Won't I be getting sequence list as size 0, I will have to add checks then, if the size is 0 then just delete the slide, else delete the references.
Else, while looping it might throw an exception?

Reply:
I will try without the flag and check.

Reply:
you can loop over an empty list (as long as the list is not null). `for (Slide slide : slides) {}` works just fine for empty lists.

Reply:
Okay